Azure-Samples/chat-with-your-data-solution-accelerator
# Add pre-commit for Bicep files

Closes #209 

## Purpose

This PR adds a local pre-commit to format Bicep files, and generate ARM templates. Note that it is quite slow, as it takes a few seconds to format and build each Bicep file. Requires AZ CLI installed.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Make updates to `.bicep` files
* Check that they have been formatted and the JSON ARM templates have been rebuilt, and added to the commit

![image](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/assets/60179183/ad1e7afa-3bf9-4f77-bb3b-ecf4784d1abe)
![image](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/assets/60179183/43193276-f85f-4565-ad76-f5ee77746357)
